### PR TITLE
Added PID loop controls and ramp rate controls

### PR DESCRIPTION
**Pull request by snyderethan**
Let me know if these functions are correct, this is what I was working on yesterday.

### CODE REVIEW

File: src/main/java/frc/robot/swerveio/AbstractSwerveModule.java [hunk 1]
```diff
@@ -68,6 +68,18 @@
...
+     * @param rate Time in seconds to go from 0 to full throttle.
+     */
+    public abstract void setPivotClosedLoopRampRate(double rate);
+
+    /**
+     * Sets the ramp rate for the drive motor.
+     * @param rate Time in seconds to go from 0 to full throttle.
+     */
+    public abstract void setDriveClosedLoopRampRate(double rate);
+
```
Comment (jordanbancino) [inline src/main/java/frc/robot/swerveio/AbstractSwerveModule.java]: What about open loop ramp rate? These are very good, now at functions for setting the drive and pivot open loop ramp rates :)

File: src/main/java/frc/robot/swerveio/NeoSwerveModule.java [hunk 2]
```diff
@@ -108,4 +108,39 @@ public void setPivotReference(double ref) {
...
+        pivotPid.setI(gain);
+    }
+
+    @Override
+    public void setPidControllerD(double gain) {
+        pivotPid.setD(gain);
+    }
+
+    @Override
+    public void setPidControllerIZone(double IZone) {
```
Comment (jordanbancino) [inline src/main/java/frc/robot/swerveio/NeoSwerveModule.java]: Variable names like this always start with a lower case. you'd want it to be `iZone` or `izone` or even just `iz`

**Comment by jordanbancino** [comment]
Nice work! You know what you're doing, for sure! There's some ambiguity in your function names though.  How do I know whether `setPidControllerP()` sets the drive PID controller or the pivot PID controller proportional value? I'd rename all of them to be something like `setPivotPidP()`, `setPivotPidI()`, etc. And of course add functions for setting the drive PID controller too. For `NeoSwerveModule` though, you can just throw `SwerveImplementationException`s because I don't think we'll be using the drive PID controller right away. For autonomous, maybe, but we aren't there yet.

**Comment by jordanbancino** [comment]
I also have to compliment you on your documentation, very well done. 👍

File: src/main/java/frc/robot/swerveio/NeoSwerveModule.java (hunk 2 above)
Comment (snyderethan) [reply src/main/java/frc/robot/swerveio/NeoSwerveModule.java]: I thought this was odd too, but the API used a capital I as well. I figured I'd just roll with it.

File: src/main/java/frc/robot/swerveio/AbstractSwerveModule.java (hunk 1 above)
Comment (snyderethan) [reply src/main/java/frc/robot/swerveio/AbstractSwerveModule.java]: Gotcha. This is what we were talking about at school today and I only had time today to figure out how to fork my own branch and put a pull request on. I'll get on this tonight and polish everything up.

**Comment by snyderethan** [comment]
I know exactly how to assign the PID Controller as well as the value assigned to it, it's already in the API as its own function but I wasn't sure if that's what we wanted. I'll do that tonight and I'll definitely rename some variables. Also, it means the world to me that you like my documentation. You're very picky.

**Pull request merged**